flytekit/common/tasks/presto_task.py

```diff
@@ -0,0 +1,157 @@
+from __future__ import absolute_import
+
+from google.protobuf.json_format import MessageToDict as _MessageToDict
+
+from flytekit.common import constants as _constants
+from flytekit.common.tasks import task as _base_task
+from flytekit.models import (
+    interface as _interface_model
+)
+from flytekit.models import literals as _literals, types as _types, \
+    task as _task_model
+
+from flytekit.common import interface as _interface
+import datetime as _datetime
+from flytekit.models import presto as _presto_models
+from flytekit.common.exceptions.user import \
+    FlyteValueException as _FlyteValueException
+from flytekit.common.exceptions import scopes as _exception_scopes
+
+
+class SdkPrestoTask(_base_task.SdkTask):
+    """
+    This class includes the logic for building a task that executes as a Presto task.
+    """
+
+    def __init__(
+            self,
+            query,
+            output_schema,
+            routing_group=None,
+            catalog=None,
+            schema=None,
+            task_inputs=None,
+            interruptible=False,
+            discoverable=False,
+            discovery_version=None,
+            retries=1,
+            timeout=None,
+    ):
+        """
+        :param Text query: Presto query specification
+        :param flytekit.common.types.schema.Schema output_schema: Schema that represents that data queried from Presto
+        :param Text routing_group: The routing group that a Presto query should be sent to for the given environment
+        :param Text catalog: The catalog to set for the given Presto query
+        :param Text schema: The schema to set for the given Presto query
+        :param dict[Text,flytekit.common.types.base_sdk_types.FlyteSdkType] task_inputs: Optional inputs to the Presto task
+        :param bool discoverable:
+        :param Text discovery_version: String describing the version for task discovery purposes
+        :param int retries: Number of retries to attempt
+        :param datetime.timedelta timeout:
+        """
+
+        # Set as class fields which are used down below to configure implicit
+        # parameters
+        self._routing_group = routing_group or ""
+        self._catalog = catalog or ""
+        self._schema = schema or ""
+
+        metadata = _task_model.TaskMetadata(
+            discoverable,
+            # This needs to have the proper version reflected in it
+            _task_model.RuntimeMetadata(
+                _task_model.RuntimeMetadata.RuntimeType.FLYTE_SDK, "1.0.0",
```

Comment:
We should use [this](https://github.com/lyft/flytekit/blob/9f3b48f2577f727f3cf1eb1292c5267d1749a3a2/flytekit/common/tasks/sdk_runnable.py#L210) for a version.